runtime: TestGdbAutotmpTypes times out on GHA Windows but is skipped upstream

## Summary

`runtime.TestGdbAutotmpTypes` times out on GitHub Actions Windows, but upstream Go's Windows builders do not actually run this test body. They skip it because `gdb` is not available on `PATH`.

This looks like an environment mismatch between upstream Windows builders and our GHA Windows runners.

## Timeout on GHA Windows

From the `windows-latest / test-std` job on March 24, 2026:

```text
panic: test timed out after 6m0s
        running tests:
                TestGdbAutotmpTypes (3m43s)
...
syscall.WaitForSingleObject(0x440, 0xffffffff)
os/exec.(*Cmd).CombinedOutput(0x3be30b4101a0)
runtime_test.TestGdbAutotmpTypes(0x3be30b2c4400)
        D:/a/happygo/happygo/go/src/runtime/runtime-gdb_test.go:626 +0x4d5
...
FAIL    runtime 360.164s
...
Terminate orphan process: pid (7400) (gdb)
Terminate orphan process: pid (6028) (gdborig)
```

## Upstream Windows behavior

LUCI / ResultDB shows Windows variants for this test, but their current results are `SKIPPED`, not `PASSED`.

Representative upstream Windows outputs:

```text
=== RUN   TestGdbAutotmpTypes
=== PAUSE TestGdbAutotmpTypes
=== CONT  TestGdbAutotmpTypes
    runtime-gdb_test.go:62: skipping: error executing gdb: exec: "gdb": executable file not found in %PATH%
--- SKIP: TestGdbAutotmpTypes (0.08s)
```

```text
=== RUN   TestGdbAutotmpTypes
=== PAUSE TestGdbAutotmpTypes
=== CONT  TestGdbAutotmpTypes
    runtime-gdb_test.go:64: skipping: error executing gdb: exec: "gdb": executable file not found in %PATH%
--- SKIP: TestGdbAutotmpTypes (0.01s)
```

ResultDB records these as `skippedReason.kind = SKIPPED_BY_TEST_BODY`.

## Relevant upstream code

`checkGdbVersion` skips if `gdb --version` cannot be executed:

```go
out, err := exec.Command("gdb", "--version").CombinedOutput()
if err != nil {
    t.Skipf("skipping: error executing gdb: %v", err)
}
```

`TestGdbAutotmpTypes` calls that first, and only later runs GDB itself:

```go
checkGdbVersion(t)
...
got, err := exec.Command("gdb", args...).CombinedOutput()
```

## Likely RCA

Upstream Windows builders do not have `gdb` on `PATH`, so the test skips early.

Our GHA Windows runners appear to have `gdb` available, so `TestGdbAutotmpTypes` proceeds into the GDB execution path and can hang long enough to blow the `runtime` package timeout.

--- a/go/src/runtime/runtime-gdb_test.go
+++ b/go/src/runtime/runtime-gdb_test.go
@@ -589,6 +589,14 @@ func TestGdbAutotmpTypes(t *testing.T) {
 	checkGdbVersion(t)
 	checkPtraceScope(t)
 
+	// NOTE(happygo): LUCI Windows builders skip this test because gdb
+	// is not on PATH (checkGdbVersion skips with "error executing gdb").
+	// However, GHA Windows runners have gdb (via MSYS2), which causes
+	// the test to run and then time out.
+	if runtime.GOOS == "windows" {
+		t.Skip("gdb tests not run on Windows upstream; times out on GHA")
+	}
+
 	if runtime.GOOS == "aix" && testing.Short() {
 		t.Skip("TestGdbAutotmpTypes is too slow on aix/ppc64")
 	}

